bin/internal/last_engine_commit.ps1

```diff
@@ -24,5 +24,48 @@ $ErrorActionPreference = "Stop"
 
 $progName = Split-Path -parent $MyInvocation.MyCommand.Definition
 $flutterRoot = (Get-Item $progName).parent.parent.FullName
+$gitToplevel = (git rev-parse --show-toplevel).Trim()
+# 1. Determine when we diverged from master.
+$MERGE_BASE_COMMIT = ""
+try {
+    $MERGE_BASE_COMMIT = (git merge-base HEAD master).Trim()
+}
+catch {
+    # If git merge-base fails (e.g., master not found, no common history),
+    # $MERGE_BASE_COMMIT will remain empty.
+}
 
-cmd /c "git log -1 --pretty=format:%H -- ""$(git rev-parse --show-toplevel)/DEPS"" ""$(git rev-parse --show-toplevel)/engine"""
+# If we did not find a merge-base, fail
+if ([string]::IsNullOrEmpty($MERGE_BASE_COMMIT)) {
+    Write-Error "Error: Could not determine a suitable engine commit." -ErrorAction Stop
+    Write-Error "Current branch: $(git rev-parse --abbrev-ref HEAD).Trim()" -ErrorAction Stop
+    Write-Error "Expected a different branch, from 'master', or a 'master' branch that exists and has history." -ErrorAction Stop
+    exit 1
+}
+
+# 2. Define and search history range to search within (unique to changes on this branch).
+$HISTORY_RANGE = "$MERGE_BASE_COMMIT..HEAD"
```

Comment:
on a release branch, the content hash will also use the release version file in internal. Not sure if you need that here or not. I guess that would be handled in the fallback on line 54.